docs/sops/serverless-controller-knative-up.md

```diff
@@ -0,0 +1,93 @@
+# OpenShift Serverless Knative Serving Control Plane Readiness
+
+## Severity: Critical
+
+## Impact
+
+Any users or services using Knative Serving will be unable to get their Knative Services reconciled.
+This means, any new Knative Services might not be created, and any existing Knative Services might not be updated. Also, it might also mean that the user workload is not being served.
+
+## Summary
+
+OpenShift Serverless does not create any alerts out-of-the box. However, users can create alerts based on the metrics exposed.
+
+The metric `knative_up{namespace="openshift-serverless", type="serving_status"}` should always have a value of `1`. 
+
+This failure should only occur in extremely catastrophic scenarios as all components are deployed in HA configuration.
+
+Possible causes can be:
+- Network issues in the cluster
+- Resource starvation on the cluster
+- OpenShift Serverless upgrade issues
+
+You should check both OpenShift Serverless operator and Knative Serving control plane status.
+
+## Steps
+
+1. Check to see if all the pods are running in `openshift-serverless` namespace:
+
+```bash
+$ oc -n openshift-serverless get pods
+```
```

Comment:
I would make this explicit: if the goal is to get all pods that are not running make the command already filter out only them like `--field-selector status.phase!=Running`

The reason of this comment is that this procedure should be as much as possible simple, for example if the command is going to return a lot of pods you might risk to don't see one of them not running

Reply:
I would prefer showing all pods.

Reason:
- Later on, there's an instruction to collect logs from pods. The `oc logs` command doesn't support something like "only logs of failing/unready/crashing pods" (it doesn't make sense to only collect the logs of those anyway. IMO, logs of the whole system should be collected)
- So, the person will collect the logs will see all logs, but they will see new pods that they haven't seen before.

We don't have too many pods anyway.

Also, the SOP I've used as a basis checks all logs, and not only the failing ones.

If one day and SRE comes and says we need to go this route, I can do it. I feel it is a bit of premature optimization at this point.